src/libPMacc/include/math/complex/Complex.hpp

```diff
@@ -0,0 +1,205 @@
+/**
+ * 2013-2015 Heiko Burau, Rene Widera, Richard Pausch, Alexander Debus
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace PMacc
+{
+namespace math
+{
+
+/** A complex number class */
+template<typename T_Type>
+struct Complex
+{
+
+public:
+
+    typedef T_Type type;
+
+    // constructor (real, imaginary)
+    HDINLINE Complex(T_Type real, T_Type imaginary = 0.0) : real(real), imaginary(imaginary) { }
+    
+    // constructor (Complex<T_OtherType>)
+    template<typename T_OtherType>
+    HDINLINE explicit Complex(const Complex<T_OtherType >& other) : real( static_cast<T_Type> (other.get_real()) ), imaginary( static_cast<T_Type> (other.get_imag()) ) { }
+
+    // default constructor ( ! no initialization of data ! )
+    HDINLINE Complex(void) { }
+
+    // Conversion from scalar (assignment)
+    HDINLINE Complex& operator=(const T_Type& other)
+    {
+        real = other;
```

Comment:
Should it be that we must also assign `this->imaginary` because we can not assume that `Complex` inizilized.
Case were a instance of `Complex` is not valid: If we use the constructor `Complex(void)`